app/frontend/helpers/ui_helper.js

```diff
@@ -216,6 +216,66 @@ class UiHelper {
     var textLoadingIndicator = this.getCurrentTextLoadingIndicator(tabIndex);
     textLoadingIndicator.hide();
   }
+
+  getFirstVisibleVerseAnchor() {
+    var firstVisibleVerseAnchor = null;
+    var verseListFrame = app_controller.getCurrentVerseListFrame();
+
+    if (verseListFrame != null) {
+      var verseListFrameRect = verseListFrame[0].getBoundingClientRect();
+
+      var currentNavigationPane = app_controller.navigation_pane.getCurrentNavigationPane()[0];
+      var currentNavigationPaneWidth = currentNavigationPane.offsetWidth;
+
+      var firstElementOffsetX = verseListFrameRect.x + currentNavigationPaneWidth + 15;
+      var firstElementOffsetY = verseListFrameRect.y + 15;
+      
+      var firstElement = document.elementFromPoint(firstElementOffsetX, firstElementOffsetY);
+      var currentElement = firstElement;
```

Comment:
I propose to declare `var currentElement` on top of the function. That way it would be understandable that it is "global" for the function scope (actually all `var`s are visible in the whole function scope. Doesn't matter where they are declared).

Reply:
currentElement is only used in the inner block of the function (within the condition `if (verseListFrame != null && verseListFrame.length > 0) {`). I usually don't put all "block-scope" variables on top.